sw/device/lib/dif/dif_keymgr.h

```diff
@@ -0,0 +1,683 @@
+   * operation errors. Hardware fault errors are caused by invalid KMAC
+   * commands, states, or outputs and are not related to software. Software
+   * operation errors are caused by invalid inputs from software. Clients can
+   * use `dif_keymgr_get_status()` to determine the type of error that occurred.
+   */
+  kDifKeymgrIrqError,
+  /**
+   * \internal Last key manager interrupt.
+   */
+  kDifKeymgrIrqLast = kDifKeymgrError,
+} dif_keymgr_irq_t;
+
+/**
+ * A snapshot of the enablement state of key manager interrupts.
+ *
+ * This is an opaque type, to be used with the `dif_keymgr_irq_disable_all()`
+ * and `dif_keymgr_irq_restore_all()` functions.
+ */
+typedef uint32_t dif_keymgr_irq_snapshot_t;
+
+/**
+ * Key manager alerts.
+ *
+ * Key manager generates alerts when it encounters a hardware or software
+ * error. Clients can use `dif_keymgr_get_status()` to determine the type of
+ * error that occurred.
+ */
+typedef enum dif_keymgr_alert {
+  /**
+   * A hardware error occurred.
+   *
+   * This alert is triggered when the hardware encounters an error condition
+   * that cannot be caused by the software, e.g. invalid KMAC commands, states,
+   * or outputs.
+   */
+  kDifKeymgrAlertHardware,
+  /**
+   * A software error occurred.
+   *
+   * This alert is triggered when the software attempts to start an invalid
+   * operation, e.g. attempting to generate keys when the key manager is at
+   * Initialized state, or use invalid inputs, e.g. a key with a forbidden
+   * version.
+   */
+  kDifKeymgrAlertSoftware,
+} dif_keymgr_alert_t;
+
+/**
+ * Key manager states.
+ *
+ * Key manager has six states that control its operation. During secure boot,
+ * key manager transitions between these states sequentially and these
+ * transitions are irreversible during a power cycle.
+ *
+ * The secret value of key manager changes at each state transition in a
+ * well-defined manner, thus its meaning is tied to the current state of key
+ * manager.
+ *
+ * The functionality of key manager is directly tied to the life cycle
+ * controller peripheral and it is explicitly disabled during specific life
+ * cycle stages. If key manager has not been initialized, it cannot be
+ * initialized until it is enabled by life cycle controller. If key manager is
+ * disabled by life cycle controller while it is in an operational state, it
+ * immediately wipes its contents and transitions to Disabled state.
+ */
+typedef enum dif_keymgr_state {
+  /**
+   * Reset state.
+   *
+   * This is the initial state of key manager after PoR. At this state, the
+   * secret value of key manager is non-deterministic, i.e. some value based on
+   * the physical characteristics of the device and environment conditions.
+   */
+  kDifKeymgrStateReset,
+  /**
+   * Initialized state.
+   *
+   * Secret value of key manager is initialized with random values generated by
+   * the entropy source.
+   */
+  kDifKeymgrStateInitialized,
```

Comment:
@tjaychen For the `Init_wip` state: why does it exist, and exposed to software? Is there some special treatment of the device we need to do while it's active? The other state transitions are asynchronous and aren't visible to software, so I'm curious about the difference.

Reply:
`Init_wip` was removed in #4467.